service/controller/resource/ensurecpcrs/create.go

```diff
@@ -45,7 +57,7 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 		}
 
 		if !exists {
-			err := r.createG8sControlPlaneCR(ctx, cr)
+			err := r.createG8sControlPlaneCR(ctx, cr, id)
```

Comment:
If the reconciliation failed between `createAWSControlPlaneCR` and `createG8sControlPlaneCR`, there's a chance `id` could be empty here. Maybe get it from the existing `AWSControlPlaneCR` in that case?

Reply:
Good shout. I added fetching the ID from the `AWSControlPlane` in this case.

Reply:
Nice!